### PR TITLE
background: Add color option

### PR DESCRIPTION
**Pull request by soreau**
This allows for setting color in the [background] section in the       
format '\\#XXXXXX' where X is a hexidecimal digit. The hashtag
must be escaped with '\\' since it's also used for comments. Fixes 26.

### CODE REVIEW

File: src/background/background.cpp [hunk 1]
```diff
@@ -289,6 +292,44 @@ WayfireBackground::reset_cycle_timeout()
...
+validate_color_string(std::string c)
+{
+    int i;
+
+    if (c.size() != 7 || c[0] != '#')
+        return false;
+
+    for (i = 1; i < 7; i++)
+    {
+        if (!isxdigit(c[i]))
```
Comment (ammen99) [inline src/background/background.cpp]: put an `std::`

File: src/background/background.cpp [hunk 2]
```diff
@@ -289,6 +292,44 @@ WayfireBackground::reset_cycle_timeout()
...
+    auto color = background_color->as_string();
+    auto style_context = window.get_style_context();
+
+    if (!validate_color_string(color))
+    {
+        std::cerr << "Ignoring invalid color: \"" << color << "\"" << std::endl;
+        std::cerr << "Color must be in format \\#XXXXXX where X is a hexidecimal digit." << std::endl;
+        color = DEFAULT_BACKGROUND_COLOR;
+    }
+std::cout << "Setting color: " << color << std::endl;
```
Comment (ammen99) [inline src/background/background.cpp]: identation

File: src/background/background.cpp (hunk 1 above)
Comment (soreau) [reply src/background/background.cpp]: Done.

File: src/background/background.cpp (hunk 2 above)
Comment (soreau) [reply src/background/background.cpp]: This was a debug message. Removed.

**Comment by ammen99** [comment]
Closing since this has been implemented in Wayfire itself.

**Pull request closed without merging**